controllers/azuremanagedmachinepool_reconciler.go

```diff
@@ -95,6 +95,8 @@ func (s *azureManagedMachinePoolService) Reconcile(ctx context.Context) error {
 	ctx, log, done := tele.StartSpanWithLogger(ctx, "controllers.azureManagedMachinePoolService.Reconcile")
 	defer done()
 
+	s.scope.SetSubnetName()
```

Comment:
Do we want to set SubnetName every time control enters this reconciles or 
can we have a check where we only set it if it is empty or if subnetName has changed ?

Reply:
Why do you think adding the check is going to be advantageous ??

Reply:
Check especially helps in scenarios when we are setting a nil pointer. (I think we ran into a problem earlier when setting `nil` struct to the _tags_. This `nil` struct updated the `resourceVersion` of the spec, and this triggered a  reconcile on every `resourceVersion` update.)

I am not completely sure if it is applicable in this scenario, but having a check shouldn't harm us.?
Let us see what others have to say for this scenario.
cc: @willie-yao @nojnhuh @CecileRobertMichon

Reply:
I agree it makes sense to only set the SubnetName if it's empty since we're trying to do defaulting here, not always ensure a value (the check should be part of the `SetSubnetName` function itself). It also improves code readability IMO: commented something like this below https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3132#discussion_r1144089256